### PR TITLE
GEOPY-1574: Modify UIJson to use pydantic structure with forms

### PR DESCRIPTION
**Pull request by benk-mira**
**GEOPY-1574 - Modify UIJson to use pydantic structure with forms**

### CODE REVIEW

**Review by domfournier — COMMENTED** [review]
See comments

File: geoh5py/ui_json/descriptors.py [hunk 1]
```diff
@@ -17,47 +17,46 @@
...
+#
+#     def __init__(self, private: str):
+#         self.private: str = private
+#
+#     def __get__(self, obj: FormParameter, objtype=None):
+#         return getattr(obj, self.private).value
+#
+#     def __set__(self, obj: FormParameter, value):
+#         getattr(obj, self.private).value = value
+#         obj._active_members.append(self.private[1:])
```
[the remark this answers is not in the document]
Comment (benk-mira) [reply geoh5py/ui_json/descriptors.py:62]: done.

File: geoh5py/ui_json/forms.py [hunk 2]
```diff
@@ -350,471 +347,471 @@ def flatten(self):
...
+#         if self.is_value:  # type: ignore # pylint: disable=no-member
+#             val = self._value.value
+#         return val
+#
+#     @value.setter
+#     def value(self, val):
+#         if isinstance(val, (int, float)):
+#             self._value.value = val
+#         else:
+#             self._property.value = val
```
[the remark this answers is not in the document]
Comment (benk-mira) [reply geoh5py/ui_json/forms.py:817]: done

File: geoh5py/ui_json/ui_json.py [hunk 3]
```diff
@@ -17,130 +17,277 @@
...
+
+    model_config: ConfigDict = ConfigDict(arbitrary_types_allowed=True)
 
-    @property
-    def validations(self):
-        """Returns a dictionary of static and inferred validations."""
+    title: str
+    geoh5: Path
+    run_command: str
+    run_command_boolean: bool
```
[the remark this answers is not in the document]
Comment (benk-mira) [reply geoh5py/ui_json/ui_json.py]: done

File: geoh5py/ui_json/ui_json.py [hunk 4]
```diff
@@ -17,130 +17,277 @@
...
+    monitoring_directory: Path
+    conda_environment: str
+    conda_environment_boolean: bool
```
[the remark this answers is not in the document]
Comment (benk-mira) [reply geoh5py/ui_json/ui_json.py]: done

File: geoh5py/ui_json/forms.py [hunk 5]
```diff
@@ -339,6 +319,23 @@ def property_if_not_is_value(self):
...
 
+    def _validate_parent(self, params: dict[str, Any]):
+        """Validate form uid is a child of the parent object."""
+        child = None
+        if isinstance(self.value, UUID):
+            child = self.value
+        elif "property" in list(self.model_fields_set) and not self.is_value:
+            child = self.property
+
+        if child is not None:
```
[the remark this answers is not in the document]
Comment (benk-mira) [reply geoh5py/ui_json/forms.py:330]: then there's no parent/child validation needed since it's just a float/int

**Review by domfournier — APPROVED** [review]

**Pull request merged**